anitya/lib/backends/github.py

```diff
@@ -50,27 +59,167 @@ def get_versions(cls, project):
         of the projects provided, project that relies on the backend of
         this plugin.
 
-        :arg Project project: a :class:`anitya.db.models.Project` object whose backend
-            corresponds to the current plugin.
-        :return: a list of all the possible releases found
-        :return type: list
-        :raise AnityaPluginException: a
-            :class:`anitya.lib.exceptions.AnityaPluginException` exception
-            when the versions cannot be retrieved correctly
+        Attributes:
+            project (:obj:`anitya.db.models.Project`): Project object whose backend
+                corresponds to the current plugin.
+
+        Returns:
+            :obj:`list`: A list of all the possible releases found
+
+        Raises:
+            AnityaPluginException: A
+                :obj:`anitya.lib.exceptions.AnityaPluginException` exception
+                when the versions cannot be retrieved correctly
 
         '''
+        owner = None
+        repo = None
         if project.version_url:
-            url_template = 'https://github.com/%(version_url)s/tags'
-            version_url = project.version_url.replace(
-                'https://github.com/', '')
-            url = url_template % {'version_url': version_url}
+            url = project.version_url
         elif project.homepage.startswith('https://github.com'):
-            url = project.homepage
+            url = project.homepage.replace('https://github.com/', '')
             if url.endswith('/'):
-                url = project.homepage[:-1]
-            url += '/tags'
+                url = url[:-1]
         else:
             raise AnityaPluginException(
                 'Project %s was incorrectly set-up' % project.name)
 
-        return get_versions_by_regex(url, REGEX, project)
+        try:
+            (owner, repo) = url.split('/')
+        except ValueError:
+            raise AnityaPluginException(
+                'Project %s was incorrectly set-up. \
+                Can\'t parse owner and repo' % project.name)
+
+        query = prepare_query(owner, repo)
+
+        try:
+            headers = REQUEST_HEADERS.copy()
+            token = config.get('GITHUB_ACCESS_TOKEN')
```

Comment:
Since you set a default for the token, you don't need to use ``get``.

Reply:
You are right